src/transport_picoquic.cpp

```diff
@@ -374,13 +376,14 @@ PicoQuicTransport::deleteStreamContext(const TransportContextId& context_id, con
         StreamContext* s_cnx = &active_streams[context_id][stream_id];
 
         on_connection_status(s_cnx, TransportStatus::Disconnected);
-        picoquic_runner_queue.push([=, cnx = s_cnx->cnx]() {
-            picoquic_close(cnx, 0);
-        });
+
+        picoquic_close(s_cnx->cnx, 0);
 
         // Remove all streams if closing the root/datagram stream (closed connection)
         active_streams.clear();
 
+        conn_context.erase(context_id);
```

Comment:
The connection context needed to be removed when stream (datagram) zero is closed. Not removing it resulted in bad access to cnx in the context.